other/upgrade.php

```diff
@@ -1839,6 +1839,13 @@ function DeleteUpgrade()
 		'upgradeData' => null,
 	);
 
+	// Fix case of Tasks directory.
+	if (is_dir(Config::$tasksdir) && is_writable(Config::$tasksdir) && is_writable(dirname(Config::$tasksdir))) {
+		rename(Config::$tasksdir, dirname(Config::$tasksdir) . DIRECTORY_SEPARATOR . 'Tasks_temp');
+		rename(dirname(Config::$tasksdir) . DIRECTORY_SEPARATOR . 'Tasks_temp', dirname(Config::$tasksdir) . DIRECTORY_SEPARATOR . 'Tasks');
+		$changes['tasksdir'] = dirname(Config::$tasksdir) . '/Tasks';
```

Comment:
I assume this should be $sourcedir?

Reply:
In 99.999% of cases, `dirname(Config::$tasksdir)` will be the same as `Config::$sourcedir`, yes. But in theory someone might have moved it, so it's best to use `dirname(Config::$tasksdir)` to be sure.

Reply:
I missed the dirname() call.

Reply:
Ah. Got it. Easy enough to overlook. 🙂

Reply:
Only thing I would suggest is maybe we check if we need to rename it or not.  I wonder if realpath check would suffice.  `realpath($taskdir) != realpath(dirname($taskdir) . '/Tasks')`?

Reply:
Also, is_dir('tasks') will not be true on case sensitive systems.  Is this intended to handle that case?

Reply:
Good idea.

Reply:
> Also, is_dir('tasks') will not be true on case sensitive systems. Is this intended to handle that case?

If `./tasks` has already been renamed to `./Tasks`, then getting false from that check is exactly what we want.

Reply:
Well not if `$tasksdir` is set to `$sorucedir . '/tasks' and we have now uploaded it as 'Tasks'.  Resulting in on case sensitive OS having both a 'tasks' and 'Tasks' directory.  Happened on one of my test boards when I manually updated it.

Reply:
Ah. Yes, now I see what you mean. Hm... that does pose a bit of a problem, doesn't it? Got any ideas?

Reply:
Not sure yet.  Windows doesn't care, it sees it as the same.  Linux will see this as if they where named 2 different things.  PHP we can use strcasecmp to check them.  Maybe `strcasecmp($taskdir) != strcasecmp(dirname($taskdir) . '/Tasks');`?

Reply:
I think I figured out a way to consistently handle all cases.

Reply:
Oh, wait, nope. I thought realpath() would report the true case of the dir's name on case insensitive file systems. Apparently that was wrong, which means I need to think about this further. Ugh.

Reply:
Got it now. This is exactly what inodes are for.